### DISCUSSION TITLE
Reorder Pages

### DISCUSSION THREAD

**Discussion by omernoach** [mozilla/pdf.js#17791] [Q&A] [answered]
Hi, Do we have the ability to reorder and remove pages?
If not, do we plan to add this feature ?

**Comment by marco-c** [comment]
Not yet, we are thinking about it but it will require quite a bit of work.

**Reply by omernoach** [reply]
cool, Thanks.
Do you know if its in short term planning?

**Reply by marco-c** [reply]
We are only thinking about it for now, it's not on our short term roadmap.

We'd certainly welcome any contributions.

**Comment by debaraj-barua** [comment]
Hi @marco-c, 

Is this feature available now? 

I stumbled onto this thread after seeing the updates from [Firefox 150](https://www.firefox.com/en-US/firefox/150.0/releasenotes/) wrt to "reorder, copy, paste, delete, and export pages".

I did not see any similar release announcement in this repo, but #21060 seemed relevant. I was just curios if this is something we can use already, or it is more of an internal api.

Cheers and thanks for all the great work in this project!

**Reply by marco-c** [reply]
Yes it is available, you can use it in Firefox already!

**Reply by debaraj-barua** [reply]
Thank you for the confirmation!! And could you let me know corresponding release for pdfjs-dist which includes this feature? Also, are there any docs/examples for this in this repo which we can look at for reference? 

The demo app here does not seem to have the same ui from Firefox, or maybe I'm missing something?

**Comment by marco-c** [comment, marked as answer]
This feature is available since Firefox 151, and with Firefox 152 we also have the ability to merge multiple PDFs into one.

**Reply by debaraj-barua** [reply]
Hi @marco-c,

Thank you for the information earlier - I wanted to follow up for some clarification:

Since the page reordering feature (as seen in recent Firefox releases) isn’t available in the [online pdf.js demo](https://mozilla.github.io/pdf.js/web/viewer.html):

- Is this feature custom-built specifically for Firefox?
- Is its implementation/code only present in the Firefox (mozilla-central) repo and not in the pdf.js repo?
- Are there any plans to expose stable APIs or official support for page reordering in pdf.js or pdfjs-dist in the future?

Thanks again for the great work, and apologies for tagging you again!

**Reply by marco-c** [reply]
@debaraj-barua the feature is in pdf.js, maybe there's an option to turn it on.
